src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs

```diff
@@ -302,6 +305,17 @@ sealed class AdoConnector {
             return stateUpdated;
         }
 
+        private async Async.Task<bool> MatchesUnlessCase(WorkItem workItem) =>
+            _config.OnDuplicate.Unless != null &&
+            await _config.OnDuplicate.Unless.ToAsyncEnumerable()
+                // Any condition from the list may match
+                .AnyAwaitAsync(async condition =>
+                    await condition.ToAsyncEnumerable()
+                        // All fields within the condition must match
+                        .AllAwaitAsync(async kvp =>
+                            workItem.Fields.TryGetValue<string>(kvp.Key, out var value) &&
+                            string.Equals(await Render(kvp.Value), value, StringComparison.OrdinalIgnoreCase)));
```

Comment:
We should probably make the `Render` funciton non-async. Looking at the implementation, we invoke `Template.RenderAsync`, but we aren't benefiting from any async IO here (we are working entirely in-memory), we could invoke `Template.Render` instead, and this would simplify all the callers.

Reply:
That's a good point, I can do that in a follow-up PR since I think it would cascade-remove a bunch async in our code.